File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;

    int area = brown + yellow;
    for (int curY = 3;curY<=area;curY++)
    {  
        int curX = area / curY;
        if(area%curY!=0) continue;
        if (brown == curX*2 + curY*2-4)
        {
            if (yellow == (curX-2) * (curY-2))
            {
                answer.push_back(curX);
                answer.push_back(curY);
                break;
            }          
        }
    }
    return answer;
}

File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    area = brown+yellow
    curX,curY = area,1

    while(1):
        if not area / curX:
            continue
        if curY <=2:
           count_brown = area 
        else:
           curX = area / curY
           count_brown = curX *2 + (curY-2)*2
        if(count_brown == brown):
            break
        curY+=1
    answer.append(curX)
    answer.append(curY)
    return answer
